src/main/java/org/alfresco/rest/framework/webscripts/ResourceWebScriptPost.java

```diff
@@ -171,7 +171,16 @@ private Object extractObjFromJson(ResourceMetadata resourceMeta, ResourceOperati
                         Object jsonContent = null;
                         if (objType != null)
                         {
-                            jsonContent = extractJsonContent(req, assistant.getJsonHelper(), objType);
+                            // check if the body is optional and is not provided
+                            if (!resourceParameter.isRequired() && Integer.valueOf(req.getHeader("content-length")) <= 0)
```

Comment:
This case should be covered by a unit test, at least.

Reply:
Methods annotated with @Operation are poorly tested by the rest api framework tests, I believe some hardening will be required in this area. Added some tests covering this context for https://issues.alfresco.com/jira/browse/REPO-2339. Since two of the feature tasks are dependent on this change for now I will merge the fix and after will add some framework contexts and tests for the same jira issue (REPO-3344).